cmd/vodloadtester/vodloadtester.go

```diff
@@ -0,0 +1,423 @@
+						} else {
+							uploadTest.TaskCheckSuccess = 1
+						}
+					}
+
+				}
+				vt.writeResultNdjson(uploadTest)
+				wg.Done()
+			}()
+		}
+		time.Sleep(vt.cliFlags.StartDelayDuration)
+	}
+
+	wg.Wait()
+
+}
+
+func (vt *vodLoadTester) importFromUrlTest(url string, runnerInfo string) {
+	wg := &sync.WaitGroup{}
+
+	for i := 0; i < int(vt.cliFlags.VideoAmount); i += int(vt.cliFlags.Simultaneous) {
+		for j := 0; j < int(vt.cliFlags.Simultaneous); j++ {
+			fmt.Printf("Importing video %d/%d\n", i+j+1, vt.cliFlags.VideoAmount)
+			wg.Add(1)
+			go func() {
+				uploadTest := uploadTest{
+					StartTime:  time.Now(),
+					RunnerInfo: runnerInfo,
+					Kind:       "import",
+				}
+
+				rndAssetName := fmt.Sprintf("load_test_import_%s", randName())
+				asset, task, err := vt.lapi.ImportAsset(url, rndAssetName)
+				defer vt.lapi.DeleteAsset(asset.ID)
+
+				if err != nil {
+					glog.Errorf("Error importing asset: %v", err)
+					uploadTest.ImportSuccess = 0
+					uploadTest.ErrorMessage = err.Error()
+					uploadTest.EndTime = time.Now()
+				} else {
+					uploadTest.AssetID = asset.ID
+					uploadTest.TaskID = task.ID
+					uploadTest.ImportSuccess = 1
+					if vt.cliFlags.TaskCheck {
+						err := vt.checkTaskProcessing(5*time.Second, api.TaskOnlyId{ID: uploadTest.TaskID})
+						if err != nil {
+							uploadTest.TaskCheckSuccess = 0
+							uploadTest.ErrorMessage = err.Error()
+						} else {
+							uploadTest.TaskCheckSuccess = 1
+						}
+					}
+
+					uploadTest.EndTime = time.Now()
+				}
+				vt.writeResultNdjson(uploadTest)
+				wg.Done()
+			}()
+		}
+		time.Sleep(vt.cliFlags.StartDelayDuration)
+	}
+	wg.Wait()
+}
+
+func (vt *vodLoadTester) writeResultNdjson(uploadTest uploadTest) {
+	uploadTest.EndTime = time.Now()
+	jsonString, err := json.Marshal(uploadTest)
+	if err != nil {
+		glog.Errorf("Error converting runTests to json: %v", err)
+	}
+	f, err := os.OpenFile(vt.cliFlags.OutputPath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		glog.Errorf("Error opening %s: %v", vt.cliFlags.OutputPath, err)
+	}
+	defer f.Close()
+	if _, err := f.WriteString(string(jsonString) + "\n"); err != nil {
+		glog.Errorf("Error writing to %s: %v", vt.cliFlags.OutputPath, err)
+	}
+}
+
+func (vt *vodLoadTester) uploadAsset(fileName string, uploadUrl string) error {
+
+	file, err := os.Open(fileName)
+
+	if err != nil {
+		fmt.Printf("error opening file %s: %v\n", fileName, err)
+		return err
+	}
+
+	err = vt.lapi.UploadAsset(vt.ctx, uploadUrl, file)
+	if err != nil {
+		fmt.Printf("error uploading asset: %v\n", err)
+		return err
+	}
+
+	return err
+}
+
+func (vt *vodLoadTester) uploadAssetResumable(url string, fileName string) error {
+
+	file, err := os.Open(fileName)
+
+	if err != nil {
+		fmt.Printf("error opening file %s: %v\n", fileName, err)
+		return err
+	}
+
+	err = vt.lapi.ResumableUpload(url, file)
+
+	if err != nil {
+		fmt.Printf("error on resumable upload asset: %v", err)
+	}
+
+	return err
+}
```

Comment:
**nit:** These are so similar, could even be a single func with an opt, WDYT?
```suggestion
func (vt *vodLoadTester) uploadAsset(fileName string, uploadUrl string, resumable bool) error {
	file, err := os.Open(fileName)
	if err != nil {
		fmt.Printf("error opening file %s: %v\n", fileName, err)
		return err
	}

        if resumable {
          err = vt.lapi.ResumableUpload(uploadUrl, file)
        } else {
	  err = vt.lapi.UploadAsset(vt.ctx, uploadUrl, file)
	}
	if err != nil {
		fmt.Printf("error uploading asset: %v\n", err)
		return err
	}

	return nil
}
```

Reply:
Good point!